streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -72,72 +72,88 @@
 
     public static class Config extends AbstractConfig {
 
-        public static Config fromStreamsConfig(StreamsConfig streamsConfig) {
+        static Config fromStreamsConfig(StreamsConfig streamsConfig) {
             return new Config(streamsConfig.originals());
         }
 
-        public Config(Map<?, ?> originals) {
+        Config(Map<?, ?> originals) {
             super(CONFIG, originals, false);
         }
-    }
 
+    }
     private final KafkaClient kafkaClient;
     private final List<MetricsReporter> reporters;
     private final Config streamsConfig;
-
+    private final Map<String, String> defaultTopicConfigs = new HashMap<>();
     private static final int MAX_INFLIGHT_REQUESTS = 100;
 
-    public StreamsKafkaClient(final StreamsConfig streamsConfig) {
-        this(Config.fromStreamsConfig(streamsConfig));
-    }
 
-    public StreamsKafkaClient(final Config streamsConfig) {
+    StreamsKafkaClient(final Config streamsConfig,
+                       final KafkaClient kafkaClient,
+                       final List<MetricsReporter> reporters) {
         this.streamsConfig = streamsConfig;
+        this.kafkaClient = kafkaClient;
+        this.reporters = reporters;
+        extractDefaultTopicConfigs(streamsConfig.originalsWithPrefix(StreamsConfig.TOPIC_PREFIX));
+    }
 
+    private void extractDefaultTopicConfigs(final Map<String, Object> configs) {
+        for (final Map.Entry<String, Object> entry : configs.entrySet()) {
+            if (entry.getValue() != null) {
```

Comment:
Could `getValue()` ever return null? `originalsWithPrefix` is fetching directly from `originals`.